p2p/net/conn.go

```diff
@@ -122,42 +123,8 @@ func (c *connectionImpl) String() string {
 // Concurrency: can be called from any go routine
 func (c *connectionImpl) Send(message []byte, id []byte) {
 	// queue messages for sending
-	c.outgoingMsgs <- OutgoingMessage{message, id}
-}
-
-// Write a message to the network connection
-// Concurrency: this message is not go safe it is designed as an internal helper to be
-// called from the event processing loop
-func (c *connectionImpl) writeMessageToConnection(om OutgoingMessage) {
```

Comment:
@avive this, the old writeMessageToConnection impl was saving op times, and message IDs.
Now we're just passing the message to the delimited(used to be msgio) chan and discard the id.

So maybe a possibility is to reimplement to code below with the delimited writer. 
or re-impl chan.go to be more taylor-made to our needs. (someone said protobuffs?) 
since it's not part of the delimited package anyway.

Reply:
*I had to comment lastOptime test for this to pass.